napari/conftest.py

```diff
@@ -450,6 +451,30 @@ def single_threaded_executor():
     executor.shutdown()
 
 
+@pytest.fixture()
+def mock_console():
```

Comment:
```suggestion
@pytest.fixture(autouse=True)
def mock_console(request):
    if "enable_console" in request.keywords:
        yield
        return
```

Maybe enable it by default and request for explicit disable?

This will require to add custom marker in `pyproject.toml` under `[tool.pytest.ini_options]`

Reply:
You mean make the fixture `autouse` and instead disable for incompatible test(s)? I think this makes sense I suspect most will want the mock in new tests.

Reply:
Yes. But we cannot use `autouse=True` here, because of headless test (without Qt). So we use `pytest_runtest_setup` for autouse